code/__HELPERS/_lists.dm

```diff
@@ -34,123 +43,75 @@
-	if(!L || !L.len || !A)
-		return FALSE
+	if(!LAZYLEN(L) || !A)
+		return
 	for(var/type in L)
 		if(istype(A, type))
 			return TRUE
-	return FALSE
 
 //Checks for specific types in specifically structured (Assoc "type" = TRUE) lists ('typecaches')
 /proc/is_type_in_typecache(atom/A, list/L)
-	if(!L || !L.len || !A)
-
-		return FALSE
-	if(ispath(A))
-		. = L[A]
-	else
-		. = L[A.type]
-
-//Checks for a string in a list
-/proc/is_string_in_list(string, list/L)
-	if(!L || !L.len || !string)
+	if(!LAZYLEN(L) || !A)
 		return
-	for(var/V in L)
-		if(string == V)
-			return TRUE
-	return
-
-//Removes a string from a list
-/proc/remove_strings_from_list(string, list/L)
-	if(!L || !L.len || !string)
-		return
-	for(var/V in L)
-		if(V == string)
-			L -= V //No return here so that it removes all strings of that type
-	return
+	return L[ispath(A) ? A : A.type]
 
 //returns a new list with only atoms that are in typecache L
 /proc/typecache_filter_list(list/atoms, list/typecache)
 	. = list()
-	for(var/thing in atoms)
-		var/atom/A = thing
-		if (typecache[A.type])
+	for(var/i in atoms)
+		var/atom/A = i
+		if(typecache[A.type])
 			. += A
 
 /proc/typecache_filter_list_reverse(list/atoms, list/typecache)
 	. = list()
-	for(var/thing in atoms)
-		var/atom/A = thing
+	for(var/i in atoms)
+		var/atom/A = i
 		if(!typecache[A.type])
 			. += A
 
 /proc/typecache_filter_multi_list_exclusion(list/atoms, list/typecache_include, list/typecache_exclude)
 	. = list()
-	for(var/thing in atoms)
-		var/atom/A = thing
+	for(var/i in atoms)
+		var/atom/A = i
 		if(typecache_include[A.type] && !typecache_exclude[A.type])
 			. += A
 
 //Like typesof() or subtypesof(), but returns a typecache instead of a list
 /proc/typecacheof(path, ignore_root_path, only_root_path = FALSE)
 	if(ispath(path))
-		var/list/types = list()
-		if(only_root_path)
-			types = list(path)
-		else
-			types = ignore_root_path ? subtypesof(path) : typesof(path)
-		var/list/L = list()
+		. = list()
+		var/list/types = only_root_path ? list(path) : ignore_root_path ? subtypesof(path) : typesof(path)
 		for(var/T in types)
-			L[T] = TRUE
-		return L
+			.[T] = TRUE
 	else if(islist(path))
-		var/list/pathlist = path
-		var/list/L = list()
+		. = list()
 		if(ignore_root_path)
-			for(var/P in pathlist)
+			for(var/P in path)
 				for(var/T in subtypesof(P))
-					L[T] = TRUE
+					.[T] = TRUE
+		else if(only_root_path)
+			for(var/P in path)
+				.[P] = TRUE
 		else
-			for(var/P in pathlist)
-				if(only_root_path)
-					L[P] = TRUE
-				else
-					for(var/T in typesof(P))
-						L[T] = TRUE
-		return L
-
-//Empties the list by setting the length to 0. Hopefully the elements get garbage collected
-/proc/clearlist(list/list)
-	if(istype(list))
-		list.len = 0
-	return
+			for(var/P in path)
+				for(var/T in typesof(P))
+					.[T] = TRUE
 
 //Removes any null entries from the list
 //Returns TRUE if the list had nulls, FALSE otherwise
 /proc/listclearnulls(list/L)

```

Comment:
You could macroize this if you removed the return value.

`L -= new /list(length(L))`

Reply:
The return value is used in some code I think

Reply:
Well that code can check the length its self and not slow shit down by making us keep this a proc

Reply:
Gonna do it